src/borg/helpers/parseformat.py

```diff
@@ -834,13 +836,27 @@ def calculate_csize(self, item):
         # note: does not support hardlink slaves, they will be csize 0
         return item.get_size(compressed=True)
 
-    def hash_item(self, hash_function, item):
-        if 'chunks' not in item:
-            return ""
+    def prepare_hash_function(self, hash_function):
         if hash_function in hashlib.algorithms_guaranteed:
             hash = hashlib.new(hash_function)
         elif hash_function == 'xxh64':
             hash = self.xxh64()
+        return hash
+
+    def hash_chunks(self, hash_function, item):
+        if 'chunks' not in item:
+            return ""
+        hash = self.prepare_hash_function(hash_function)
+        for chunk in item.chunks:
+            hash.update(chunk.id)
+            hash.update(bytes(chunk.size))
+            hash.update(bytes(chunk.csize))
```

Comment:
did you copy that from somewhere? can you give a link to that place?

Reply:
If you mean "where did I get the attributes to hash?" I looked at repr(chunk) and included everything.

Reply:
i thought we maybe have code like that already somewhere.

but maybe we don't and it only compared the full lists, not hashes of the lists.

btw, guess it would be enough to only update the hash with the chunk.id - as the id is a cryptographic hash already, adding the size and csize also is maybe superfluous or (in case of csize) even harmful.